### PR TITLE
Dotnet masterclass week4 homework

### PR DESCRIPTION
**Pull request by amritpandey**

### CODE REVIEW

File: week2/exercise/Program.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        sum = sum + n % 10;
+        n = n / 10;
+    }
+    return sum;
+
+}
+
+int CountCapitalLetters(string input)
+{
+    {
```
Comment (cduplanday) [inline week2/exercise/Program.cs:92]: It is enough to have a single pair of `{` here

File: week4/homework/Program.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+Price (this simulates creating a product)
+*/
+app.MapPost("/createProduct", async (Product productData) =>
+{
+    var httpClient = new HttpClient();
+    var response= await httpClient.PostAsJsonAsync("https://dummyjson.com/products/add", productData)
+        ;
+        System.Console.WriteLine(response);
+        var data = await response.Content.ReadFromJsonAsync<Product>();
+        var responseData = new
```
Comment (cduplanday) [inline week4/homework/Program.cs]: If you want to improve this even more, you can create a new class or record for this so you avoid using the object
```
  var postResponce = new
  {
    StatusCode = postRequest.StatusCode,
    ReasonPhrase = postRequest.ReasonPhrase,
    data = UserDataResponce,
    Date = postRequest.Headers.Date,
  };
```

then you could have the Data set as a Generic type and then you can use the same class for all of the other responses.
Please reach out if you want to do this and need assistance

File: week4/homework/Program.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+app.MapPost("/retrieveUsers", async (UserId ids) =>
+{
+    var httpClient = new HttpClient();
+        var result = "";
+        foreach (var id in ids.Id)
+        
+        {
+            var response= await httpClient.GetAsync($"https://dummyjson.com/users/{id}");
+            var data = await response.Content.ReadFromJsonAsync<User>();
+             result += data.FirstName +" "+ data.LastName+" "+data.Age +" ";
```
Comment (cduplanday) [inline week4/homework/Program.cs]: You could create a list of users here and return that in your endpoint. It is more common to return a list of items as a response than a string or an anonymous object.

**Review by cduplanday — APPROVED** [review]
Good job!
I left a few comments, please reach out if you have questions.

File: week4/homework/Program.cs (hunk 2 above)
Comment (amritpandey) [reply week4/homework/Program.cs]: Hi I made a new class but unable to find the datatype for those parameters. Could you please help me with this.

Comment (cduplanday) [reply week4/homework/Program.cs]: Sorry for the late response. 
you can create a record:
```
public record ApiResponse<T>(HttpStatusCode StatusCode, string ReasonPhrase, T data);
```

then you use it like so
```
var responseData = new ApiResponse<User>(response.StatusCode, response.ReasonPhrase, data);
```

File: week4/homework/Program.cs (hunk 3 above)
Comment (amritpandey) [reply week4/homework/Program.cs]: yes I made a new userlist and return a list of items as a response. Would like to get your feedback.

File: week4/homework/Program.cs (hunk 2 above)
Comment (amritpandey) [reply week4/homework/Program.cs]: Implemented and thanks

**Pull request merged**